code/main/media/mediaModule.brs

```diff
@@ -0,0 +1,245 @@
+    end if
+
+    if _adb_isIdentityModule(identityModule) = false then
+        _adb_logError("_adb_EdgeModule() - identityModule is not valid.")
+        return invalid
+    end if
+
+    module = _adb_AdobeObject("com.adobe.module.media")
+
+    module.Append({
+        ' constants
+        _CONSTANTS: _adb_InternalConstants(),
+
+        ' external dependencies
+        _configurationModule: configurationModule,
+        _identityModule: identityModule,
+        _edgeRequestWorker: _adb_EdgeRequestWorker(),
+        _sessionManager: _adb_MediaSessionManager(),
+
+        ' Event data example:
+        ' {
+        '     clientSessionId: "xx-xxxx-xxxx",
+        '     tsObject: { ts_inISO8601: "2019-01-01T00:00:00.000Z", ts_inMillis: 1546300800000 },
+        '     xdmData: { xdm: {} }
+        '     configuration: { }
+        ' }
+        processEvent: sub(requestId as string, eventData as object)
+            mediaEventType = eventData.xdmData.xdm.eventType
+            clientSessionId = eventData.clientSessionId
+            tsObject = eventData.tsObject
+            xdmData = eventData.xdmData
+
+            if mediaEventType = m._CONSTANTS.MEDIA.SESSION_START_EVENT_TYPE
+                sessionConfig = eventData.configuration
+                m._sessionStart(clientSessionId, sessionConfig, xdmData, tsObject)
+                return
+            end if
+
+            if mediaEventType <> invalid and _adb_isStringInArray(mediaEventType, m._CONSTANTS.MEDIA.EVENT_TYPES)
+                m._actionInSession(requestId, clientSessionId, xdmData, tsObject)
+            else
+                _adb_logError("processEvent() - media event type is invalid: " + FormatJson(mediaEventType))
+            end if
+        end sub,
+
+        _mediaConfigIsNotReady: function() as boolean
+            ' appVersion is optional for sessionStart request
+            return _adb_isEmptyOrInvalidString(m._configurationModule.getMediaChannel()) or _adb_isEmptyOrInvalidString(m._configurationModule.getMediaPlayerName())
+        end function,
+
+        _sessionStart: sub(clientSessionId as string, sessionConfig as object, xdmData as object, tsObject as object)
+            ' TODO: the session-level config should be merged with the global config, before the validation
+
+            if m._mediaConfigIsNotReady() then
+                _adb_logError("_sessionStart() - the media session is not created/started properly (missing the channel name or the player name).")
+                return
+            end if
+
+            m._sessionManager.createNewSession(clientSessionId)
+            ' TODO: add session-level config to the sessionManager
+            ' m._sessionManager.createNewSession(sessionConfig,sessionConfig["config.mainpinginterval"],sessionConfig["config.adpinginterval"])
+
+            appVersion = m._configurationModule.getMediaAppVersion()
+            xdmData.xdm["_id"] = _adb_generate_UUID()
+            xdmData.xdm["timestamp"] = tsObject.ts_inISO8601
```

Comment:
We should check if timestamp is provided by the customer.

Reply:
The same question as we discussed. I'd suggest not asking customers to provide a timestamp value because It's hard to convert the ISO8601 string to the time stamp in milliseconds format.